### DISCUSSION TITLE
Changes to Buffer Size for Sunbird video player

### DISCUSSION THREAD

**Discussion by pushpamtrigyn** [Sunbird-Knowlg/Community#169] [General Discussions]
Hi,

We had been testing the sunbird video player which is connected to Oracle streaming services for DIKSHA. However, due to certain issues while playing the video some errors start to come while rendering them. Oracle team suggested some changes to the **buffer configuration** which we had provided to them (specifically videoJS) :-

var Config = {
    GOAL_BUFFER_LENGTH: 20, // reduced from 30
    MAX_GOAL_BUFFER_LENGTH: 40, // reduced from 60
    BACK_BUFFER_LENGTH: 20, // reduced from 30
    GOAL_BUFFER_LENGTH_RATE: 1,
    INITIAL_BANDWIDTH: 2097152, // reduced to 2 MB/s
    BANDWIDTH_VARIANCE: 1.3, // slightly increased
    BUFFER_LOW_WATER_LINE: 5, // increased from 0
    MAX_BUFFER_LOW_WATER_LINE: 20, // reduced from 30
    EXPERIMENTAL_MAX_BUFFER_LOW_WATER_LINE: 16,
    BUFFER_LOW_WATER_LINE_RATE: 1,
    BUFFER_HIGH_WATER_LINE: 20 // reduced from 30
};

Although we have modified them in sunbird video player and published our own package, the changes does not seem to be reflected. Is there any other way to explicitly set them in the codebase ? We modified these at the following paths before publishing our own npm package:-

<img width="458" alt="image" src="https://github.com/Sunbird-Knowlg/Community/assets/127098111/4f0f6b50-f1e4-4dba-ab70-4e3031ba8da0">

Instances of videoJS are also found in player (Sunbird ED portal) but we are not sure whether to modify them.

Any leads and guidance on the same will be fruitful for us. Thanks !!

CC - @vinukumar-vs @Krishnaj20 @samarthtrigyn

**Comment by HarishGangula** [comment]
@pushpamtrigyn  are these changes of the config updated in video js and are we using that modified videojs as a dependency in Sunbird video player since video js is a dependency it might get replaced. Check this and let us know.
cc: @pallakartheekreddy

**Reply by pushpamtrigyn** [reply]
@HarishGangula We had modified the configuration in the build files of VideoJS and then created an npm package for sunbird video player (posted in previous post screenshot).

Also, there is a folder called "web-component" and has a file "sunbird-video-player.js" where we changed it (I think that is actually used for the local setup of sunbird video player rather than using in player itself)

**Reply by HarishGangula** [reply]
@pushpamtrigyn Actually videojs is a peer dependency from the sunbird video player so if you are using the sunbird video player as Angular library in another angular app you have to do the changes in the angular app's node modules folder and check once.

web component file is not for local but to use the sunbird video player as web component in other framework or simple js projects.

Please check the documentation for same in readme file of the player.

cc: @pallakartheekreddy

**Reply by samarthtrigyn** [reply]
Hi @HarishGangula So you are saying that the there is no configuration which we should change for the videojs player, but we need to change the buffer configurations of the sunbird video player which internally will use the videojs player as a dependency for playing videos?

If we want to change the buffer configurations of the sunbird video player then how and where can we do that?
cc @pushpamtrigyn

**Reply by samarthnigam** [reply]
Hi @HarishGangula is it possible to have a virtual meet to see this please. we are stuck at the same place for a few days now? 
cc @AmiableAnil @pushpamtrigyn

**Reply by HarishGangula** [reply]
Yes @samarthnigam  we can have a meeting today at 5 PM please include me and @pallakartheekreddy

**Reply by samarthnigam** [reply]
Hi @HarishGangula requesting your time for tommorow please. I missed sending the invite for today.

**Reply by HarishGangula** [reply]
@samarthnigam  Can we connect today at 12:30 PM

**Reply by samarthnigam** [reply]
Yes we can. Let me share the meeting link.

**Reply by samarthnigam** [reply]
Hi @HarishGangula Can you please join here on zoom

**Comment by samarthtrigyn** [comment]
Yes we can.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by HarishGangula** [comment]
@samarthnigam  As we discussed , it's crucial to update the configuration at the appropriate location to observe the desired changes in player behavior for streaming. Based on your implementation, you have the option to configure either within the player or within the container (portal/mobile app).

cc: @pallakartheekreddy  @mohitga

**Reply by pallakartheekreddy** [reply]
@samarthnigam As we haven't received a response from you, we will proceed to close this thread. If you have any further questions or require assistance in the future, please don't hesitate to start a new thread. Thank you.
